src/lib.rs

```diff
@@ -179,60 +192,63 @@ fn camera_movement_system(
 	}
 }
 
-#[derive(Default)]
-struct State {
-	mouse_motion_event_reader: EventReader<MouseMotion>,
-}
-
 fn mouse_motion_system(
 	time: Res<Time>,
-	mut state: ResMut<State>,
-	mouse_motion_events: Res<Events<MouseMotion>>,
+	mut mouse_motion_events: EventReader<MouseMotion>,
+	mut mouse_scroll_events: EventReader<MouseWheel>,
 	mut query: Query<(&mut FlyCamera, &mut Transform)>,
 ) {
-	let mut delta: Vec2 = Vec2::zero();
-	for event in state.mouse_motion_event_reader.iter(&mouse_motion_events) {
+	let mut delta = Vec2::ZERO;
+	for event in mouse_motion_events.iter() {
 		delta += event.delta;
 	}
 	if delta.is_nan() {
 		return;
 	}
 
+	let scroll_delta = mouse_scroll_events
+		.iter()
+		.fold(0.0, |acc, event| acc + event.y);
+
 	for (mut options, mut transform) in query.iter_mut() {
+		if scroll_delta < 0.0 {
+			options.speed *= 1.0 - options.scroll_sensitivity;
```

Comment:
The options took me a while to figure out what it is, that's why consider renaming it to fly_camera.